### ISSUE TITLE
burnin fails if test VM hostname does not resolve

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#887]
Originally reported of Google Code with ID 831.

```
[ganeti 2.10.3 under Debian Wheezy, installed from backports]

Although burnin has "--no-name-check" and "--no-ip-check" options, it fails at the "rename" step if the given hostname does not resolve.

# /usr/lib/ganeti/tools/burnin -H kvm:kernel_path=/vmlinuz,initrd_path=/initrd.img \
  -o debootstrap+default --disk-size 1024m --no-name-check --no-ip-check testvm
...
- Renaming the instances to their own name
  * instance testvm
    rename to the same name
- Idempotent opcode failed, retry #1/3: ('The given name (testvm) does not resolve: Name or service not known', 'resolver_error')
- Idempotent opcode failed, retry #2/3: ('The given name (testvm) does not resolve: Name or service not known', 'resolver_error')
- Idempotent opcode repeated failure, aborting
- Error detected: opcode buffer follows:


- Sat May 17 04:47:05 2014 * creating instance disks...
Sat May 17 04:47:08 2014 adding instance testvm to cluster config
Sat May 17 04:47:08 2014  - INFO: Waiting for instance testvm to sync disks
Sat May 17 04:47:08 2014  - INFO: - device disk/0:  0.80% done, 4m 50s remaining (estimated)
Sat May 17 04:48:08 2014  - INFO: Instance testvm's disks are in sync
Sat May 17 04:48:08 2014 * running the instance OS create scripts...
Sat May 17 04:54:45 2014 * starting instance...
Sat May 17 04:54:51 2014 Replacing disk(s) 0 for instance 'testvm'
Sat May 17 04:54:51 2014 Current primary node: host19.ws.nsrc.org
Sat May 17 04:54:51 2014 Current seconary node: host20.ws.nsrc.org
Sat May 17 04:54:51 2014 STEP 1/6 Check device existence
Sat May 17 04:54:51 2014  - INFO: Checking disk/0 on host19.ws.nsrc.org
Sat May 17 04:54:51 2014  - INFO: Checking disk/0 on host20.ws.nsrc.org
Sat May 17 04:54:51 2014  - INFO: Checking volume groups
Sat May 17 04:54:51 2014 STEP 2/6 Check peer consistency
Sat May 17 04:54:51 2014  - INFO: Checking disk/0 consistency on node host19.ws.nsrc.org
Sat May 17 04:54:52 2014 STEP 3/6 Allocate new storage
Sat May 17 04:54:52 2014  - INFO: Adding storage on host20.ws.nsrc.org for disk/0
Sat May 17 04:54:52 2014 STEP 4/6 Changing drbd configuration
Sat May 17 04:54:52 2014  - INFO: Detaching disk/0 drbd from local storage
Sat May 17 04:54:53 2014  - INFO: Renaming the old LVs on the target node
Sat May 17 04:54:53 2014  - INFO: Renaming the new LVs on the target node
Sat May 17 04:54:53 2014  - INFO: Adding new mirror component on host20.ws.nsrc.org
Sat May 17 04:54:54 2014 STEP 5/6 Sync devices
Sat May 17 04:54:54 2014  - INFO: Waiting for instance testvm to sync disks
Sat May 17 04:54:55 2014  - INFO: - device disk/0:  1.20% done, 3m 57s remaining (estimated)
Sat May 17 04:55:55 2014  - INFO: Instance testvm's disks are in sync
Sat May 17 04:55:55 2014 STEP 6/6 Removing old storage
Sat May 17 04:55:55 2014  - INFO: Remove logical volumes for disk/0
Sat May 17 04:55:56 2014 Replacing disk(s) 0 for instance 'testvm'
Sat May 17 04:55:56 2014 Current primary node: host19.ws.nsrc.org
Sat May 17 04:55:56 2014 Current seconary node: host20.ws.nsrc.org
Sat May 17 04:55:56 2014 STEP 1/6 Check device existence
Sat May 17 04:55:56 2014  - INFO: Checking disk/0 on host20.ws.nsrc.org
Sat May 17 04:55:56 2014  - INFO: Checking disk/0 on host19.ws.nsrc.org
Sat May 17 04:55:56 2014  - INFO: Checking volume groups
Sat May 17 04:55:57 2014 STEP 2/6 Check peer consistency
Sat May 17 04:55:57 2014  - INFO: Checking disk/0 consistency on node host20.ws.nsrc.org
Sat May 17 04:55:57 2014 STEP 3/6 Allocate new storage
Sat May 17 04:55:57 2014  - INFO: Adding storage on host19.ws.nsrc.org for disk/0
Sat May 17 04:55:58 2014 STEP 4/6 Changing drbd configuration
Sat May 17 04:55:58 2014  - INFO: Detaching disk/0 drbd from local storage
Sat May 17 04:55:58 2014  - INFO: Renaming the old LVs on the target node
Sat May 17 04:55:58 2014  - INFO: Renaming the new LVs on the target node
Sat May 17 04:55:59 2014  - INFO: Adding new mirror component on host19.ws.nsrc.org
Sat May 17 04:55:59 2014 STEP 5/6 Sync devices
Sat May 17 04:55:59 2014  - INFO: Waiting for instance testvm to sync disks
Sat May 17 04:56:00 2014  - INFO: - device disk/0:  1.60% done, 1m 23s remaining (estimated)
Sat May 17 04:57:00 2014  - INFO: Instance testvm's disks are in sync
Sat May 17 04:57:00 2014 STEP 6/6 Removing old storage
Sat May 17 04:57:00 2014  - INFO: Remove logical volumes for disk/0
Sat May 17 04:57:01 2014 Replacing disk(s) 0 for instance 'testvm'
Sat May 17 04:57:01 2014 Current primary node: host19.ws.nsrc.org
Sat May 17 04:57:01 2014 Current seconary node: host20.ws.nsrc.org
Sat May 17 04:57:01 2014 STEP 1/6 Check device existence
Sat May 17 04:57:01 2014  - INFO: Checking disk/0 on host19.ws.nsrc.org
Sat May 17 04:57:01 2014  - INFO: Checking volume groups
Sat May 17 04:57:01 2014 STEP 2/6 Check peer consistency
Sat May 17 04:57:01 2014  - INFO: Checking disk/0 consistency on node host19.ws.nsrc.org
Sat May 17 04:57:02 2014 STEP 3/6 Allocate new storage
Sat May 17 04:57:02 2014  - INFO: Adding new local storage on host21.ws.nsrc.org for disk/0
Sat May 17 04:57:02 2014 STEP 4/6 Changing drbd configuration
Sat May 17 04:57:02 2014  - INFO: activating a new drbd on host21.ws.nsrc.org for disk/0
Sat May 17 04:57:03 2014  - INFO: Shutting down drbd for disk/0 on old node
Sat May 17 04:57:03 2014  - INFO: Detaching primary drbds from the network (=> standalone)
Sat May 17 04:57:03 2014  - INFO: Updating instance configuration
Sat May 17 04:57:03 2014  - INFO: Attaching primary drbds to new secondary (standalone => connected)
Sat May 17 04:57:04 2014 STEP 5/6 Sync devices
Sat May 17 04:57:04 2014  - INFO: Waiting for instance testvm to sync disks
Sat May 17 04:57:04 2014  - INFO: - device disk/0:  1.20% done, 2m 10s remaining (estimated)
Sat May 17 04:58:05 2014  - INFO: Instance testvm's disks are in sync
Sat May 17 04:58:05 2014 STEP 6/6 Removing old storage
Sat May 17 04:58:05 2014  - INFO: Remove logical volumes for 0
Sat May 17 04:58:07 2014 Growing disk 0 of instance 'testvm' by 128M to 1.1G
Sat May 17 04:58:08 2014  - INFO: Waiting for instance testvm to sync disks
Sat May 17 04:58:08 2014  - INFO: - device disk/0:  9.10% done, 17s remaining (estimated)
Sat May 17 04:58:25 2014  - INFO: Instance testvm's disks are in sync
Sat May 17 04:58:26 2014 Failover instance testvm
Sat May 17 04:58:26 2014 * checking disk consistency between source and target
Sat May 17 04:58:26 2014 * shutting down instance on source node
Sat May 17 05:00:30 2014 * deactivating the instance's disks on source node
Sat May 17 05:00:30 2014 * activating the instance's disks on target node host21.ws.nsrc.org
Sat May 17 05:00:31 2014 * starting the instance on the target node host21.ws.nsrc.org
Sat May 17 05:00:34 2014 Migrating instance testvm
Sat May 17 05:00:35 2014 * checking disk consistency between source and target
Sat May 17 05:00:35 2014 * switching node host19.ws.nsrc.org to secondary mode
Sat May 17 05:00:35 2014 * changing into standalone mode
Sat May 17 05:00:35 2014 * changing disks into dual-master mode
Sat May 17 05:00:36 2014 * wait until resync is done
Sat May 17 05:00:36 2014 * preparing host19.ws.nsrc.org to accept the instance
Sat May 17 05:00:38 2014 * migrating instance to host19.ws.nsrc.org
Sat May 17 05:00:39 2014 * starting memory transfer
Sat May 17 05:00:42 2014 * memory transfer complete
Sat May 17 05:00:42 2014 * switching node host21.ws.nsrc.org to secondary mode
Sat May 17 05:00:42 2014 * wait until resync is done
Sat May 17 05:00:42 2014 * changing into standalone mode
Sat May 17 05:00:42 2014 * changing disks into single-master mode
Sat May 17 05:00:44 2014 * wait until resync is done
Sat May 17 05:00:44 2014 * done
Sat May 17 05:00:44 2014  - INFO: Not checking memory on the secondary node as instance will not be started
Sat May 17 05:00:45 2014 Migrating instance testvm
Sat May 17 05:00:45 2014 * checking where the instance actually runs (if this hangs, the hypervisor might be in a bad state)
Sat May 17 05:00:45 2014 * instance confirmed to be running on its primary node (host19.ws.nsrc.org)
Sat May 17 05:00:45 2014 * switching node host21.ws.nsrc.org to secondary mode
Sat May 17 05:00:45 2014 * wait until resync is done
Sat May 17 05:00:45 2014 * changing into standalone mode
Sat May 17 05:00:46 2014 * changing disks into single-master mode
Sat May 17 05:00:47 2014 * wait until resync is done
Sat May 17 05:00:47 2014 * done
Sat May 17 05:00:48 2014 Shutting down instance testvm
Sat May 17 05:02:52 2014 Creating a snapshot of disk/0 on node host19.ws.nsrc.org
Sat May 17 05:02:52 2014 Starting instance testvm
Sat May 17 05:02:54 2014 Exporting snapshot/0 from host19.ws.nsrc.org to host21.ws.nsrc.org
Sat May 17 05:02:57 2014 snapshot/0 is now listening, starting export
Sat May 17 05:03:00 2014 snapshot/0 is receiving data on host21.ws.nsrc.org
Sat May 17 05:03:00 2014 snapshot/0 is sending data on host19.ws.nsrc.org
Sat May 17 05:03:05 2014 snapshot/0 sent 95M, 16.0 MiB/s
Sat May 17 05:03:15 2014 snapshot/0 finished receiving data
Sat May 17 05:03:15 2014 snapshot/0 finished sending data
Sat May 17 05:03:15 2014 Removing snapshot of disk/0 on node host19.ws.nsrc.org
Sat May 17 05:03:15 2014 Finalizing export on host21.ws.nsrc.org
Sat May 17 05:03:16 2014 Removing old exports for instance testvm
Sat May 17 05:05:22 2014 * creating instance disks...
Sat May 17 05:05:24 2014 adding instance testvm to cluster config
Sat May 17 05:05:25 2014  - INFO: Waiting for instance testvm to sync disks
Sat May 17 05:05:25 2014  - INFO: - device disk/0:  1.60% done, 1m 32s remaining (estimated)
Sat May 17 05:06:25 2014  - INFO: Instance testvm's disks are in sync
Sat May 17 05:06:25 2014 * running the instance OS import scripts...
Sat May 17 05:06:25 2014 Exporting disk/0 from host21.ws.nsrc.org to host19.ws.nsrc.org
Sat May 17 05:06:28 2014 disk/0 is now listening, starting export
Sat May 17 05:06:31 2014 disk/0 is receiving data on host19.ws.nsrc.org
Sat May 17 05:06:31 2014 disk/0 is sending data on host21.ws.nsrc.org
Sat May 17 05:06:36 2014 disk/0 sent 106M, 17.8 MiB/s, 61%, ETA 4s
Sat May 17 05:06:41 2014 disk/0 finished receiving data
Sat May 17 05:06:41 2014 disk/0 finished sending data
Sat May 17 05:06:41 2014 Running rename script for testvm
Sat May 17 05:06:42 2014 * starting instance...
Sat May 17 05:06:44 2014  - WARNING: Failed to query node host20.ws.nsrc.org (continuing): No exports directory
Sat May 17 05:08:51 2014 Running the instance OS create scripts...
Sat May 17 05:08:56 2014 Changing OS to 'debootstrap+default'...
Sat May 17 05:08:57 2014 Running the instance OS create scripts...

-


- Removing instances
  * instance testvm
Traceback (most recent call last):
  File "/usr/lib/ganeti/tools/burnin", line 21, in <module>
    sys.exit(main.Main())
  File "/usr/share/ganeti/2.10/ganeti/tools/burnin.py", line 1165, in Main
    return Burner().BurninCluster()
  File "/usr/share/ganeti/2.10/ganeti/tools/burnin.py", line 1113, in BurninCluster
    self.BurnRenameSame()
  File "/usr/share/ganeti/2.10/ganeti/tools/burnin.py", line 274, in wrapper
    val = fn(self, *args, **kwargs)
  File "/usr/share/ganeti/2.10/ganeti/tools/burnin.py", line 294, in batched
    val = fn(self, *args, **kwargs)
  File "/usr/share/ganeti/2.10/ganeti/tools/burnin.py", line 910, in BurnRenameSame
    self.ExecOrQueue(instance, [op1, op2, op4])
  File "/usr/share/ganeti/2.10/ganeti/tools/burnin.py", line 402, in ExecOrQueue
    val = self.ExecOp(self.queue_retry, *ops) # pylint: disable=W0142
  File "/usr/share/ganeti/2.10/ganeti/tools/burnin.py", line 394, in ExecOp
    return self.MaybeRetry(rval, "opcode", self._ExecOp, *ops)
  File "/usr/share/ganeti/2.10/ganeti/tools/burnin.py", line 366, in MaybeRetry
    self.MaybeRetry(retry_count - 1, msg, fn, *args)
  File "/usr/share/ganeti/2.10/ganeti/tools/burnin.py", line 366, in MaybeRetry
    self.MaybeRetry(retry_count - 1, msg, fn, *args)
  File "/usr/share/ganeti/2.10/ganeti/tools/burnin.py", line 351, in MaybeRetry
    val = fn(*args)
  File "/usr/share/ganeti/2.10/ganeti/tools/burnin.py", line 376, in _ExecOp
    results = cli.PollJob(job_id, cl=self.cl, feedback_fn=self.Feedback)
  File "/usr/share/ganeti/2.10/ganeti/cli.py", line 2265, in PollJob
    return GenericPollJob(job_id, _LuxiJobPollCb(cl), reporter)
  File "/usr/share/ganeti/2.10/ganeti/cli.py", line 2087, in GenericPollJob
    errors.MaybeRaise(msg)
  File "/usr/share/ganeti/2.10/ganeti/errors.py", line 501, in MaybeRaise
    raise errcls(*args)
ganeti.errors.OpPrereqError: ('The given name (testvm) does not resolve: Name or service not known', 'resolver_error')
```
Originally added on 2014-05-17 05:25:56 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-05-20 12:29:53 +0000 UTC.

Added Labels: Type-Defect Priority-High
Changed State: Accepted
Added to Milestone: Release2.13

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed with:

    Make burnin respect --no-ip-check and --no-name-check
    
    This patch fixes issue 831. Burnin ignores the
    --no-ip-check and --no-name-checks when renaming
    instances and thus fails due to not being able to
    resolve names that are not really FQDN.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Jose Lopes <jabolopes@google.com>

```
Originally added on 2014-05-20 13:10:08 +0000 UTC.

Mentioned Issue with GoogleCode ID 831 has been migrated to GitHub with ID #887.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:37:02 +0000 UTC.

Changed State: Released
